src/runmode-af-packet.c

```diff
@@ -222,6 +222,18 @@ void *ParseAFPConfig(const char *iface)
                 aconf->iface);
         aconf->flags |= AFP_RING_MODE;
     }
+    (void)ConfGetChildValueBoolWithDefault(if_root, if_default, "tpacket-v3", (int *)&boolval);
+    if (boolval) {
+        if (strcasecmp(RunmodeGetActive(), "workers") == 0) {
+            SCLogInfo("Enabling tpacket v3 capture on iface %s",
+                    aconf->iface);
+            aconf->flags |= AFP_TPACKET_V3;
+        } else {
+            SCLogError(SC_ERR_RUNMODE,
```

Comment:
Problem is that v3 release a block based so it would meant we maintain a reference count of the block and release the block once we have no more reference. We would need at least one atomic to do so. So it is doable but it may be a bit complicated. Alternative would be to forget about zero copy in non workers runmode.